shibata added English-version file of the difference after February 10 (#110)
fujitake/vantiq-related: vantiq-external-services-integration/docs/en/vantiq-azure-CosmosDB.md

# Introduction
Describe how to collaborate with Azure Cosmos DB APIs directly from Vantiq.  

> In this document, describe how to do the following from Vantiq's Procedures to Cosmos DB.  

Reviewer: ```suggestion
This document describes how to do the following from Vantiq's Procedures to Cosmos DB.  

```

As for the details of generation method, refer to [Access control in the Azure Cosmos DB SQL API - Azure Cosmos DB REST API | Microsoft Docs](https://docs.microsoft.com/en-us/rest/api/cosmos-db/access-control-on-cosmosdb-resources).  

```
PROCEDURE getAuthorizationTokenUsingMasterKey(verb, resourceType, resourceId, date, masterKey)

var TYPE = "master"
var VERSION = "1.0"

var key = Decode.base64Raw(masterKey)

var text = ""
text += verb ? (verb.toLowerCase() + "\n") : "\n"
text += resourceType ? (resourceType.toLowerCase() + "\n") : "\n"
text += resourceId ? (resourceId + "\n") : "\n"
text += date ? (date.toLowerCase() + "\n") : "\n"
text += "\n"

var signature = Hash.hmacSha256(key, text)
signature = Encode.base64(signature)
var token = "type=" + TYPE + "&ver=" + VERSION + "&sig=" + signature
token = encodeUriComponent(token)
return token

```

This Procedure will be called within the Procedure that makes the following API call.  

### Create Document

Create the Procedure that creates a Document (registers records to Cosmos DB).  
Reference: [Create Document - Azure Cosmos DB REST API | Microsoft Docs](https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-document)  

> It is necessary to specify an ID that does not duplicate when creating a document to CosmosDB.    

Reviewer: ```suggestion
It is necessary to specify an ID that is unique when creating a document in CosmosDB.    

```

var SENSOR_ID = "sensor_1"
var TEMPERATURE = 17.2

var now = now()
var date = format("{0, date,EEE',' dd MMM yyyy HH:mm:ss 'GMT'}",now)

var token = getAuthorizationTokenUsingMasterKey(REQUEST_METHOD, RESOURCE_TYPE, RESOURCE_ID, date, MASTER_KEY)

var body = {
    "id": SENSOR_ID + "_" + toString(date(now, "date", "epochMilliseconds")),
    "sensor_id": SENSOR_ID,
    "temperature": TEMPERATURE
}

var header = {
    "Accept": "application/json",
    "Authorization": token,
    "x-ms-date": date,
    "x-ms-version": "2018-12-31",
    "x-ms-documentdb-partitionkey": "[\"" + SENSOR_ID + "\"]"  
}

var requestPath = "/" + RESOURCE_ID + "/docs"

SELECT * FROM SOURCE CosmosDB WITH method = REQUEST_METHOD, headers = header, body = body, path = requestPath

```

Executing the Procedure, the following results are returned.  

```json
[
   {
      "id": "sensor_1_1650619266361",
      "sensor_id": "sensor_1",
      "temperature": 17.2,
      "_rid": "02MwAIrnwS0OAAAAAAAAAA==",
      "_self": "dbs/02MwAA==/colls/02MwAIrnwS0=/docs/02MwAIrnwS0OAAAAAAAAAA==/",
      "_etag": "\"0200300c-0000-0700-0000-626273820000\"",
      "_attachments": "attachments/",
      "_ts": 1650619266
   }
]
```

> From Azure Portal, it is also possible to confirm that the above document is stored in the Items of the sensors container that was created.  

Reviewer: ```suggestion
From Azure Portal, it is also possible to confirm that the above document is stored in the Items of the sensors container.  

```
